File: app/src/main/java/de/bikebean/app/db/type/types/sms_parser_types/WifiOff.java
package de.bikebean.app.db.type.types.sms_parser_types;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.lang.ref.WeakReference;

import de.bikebean.app.db.settings.settings.Battery;
import de.bikebean.app.db.settings.settings.replace_if_newer_settings.Status;
import de.bikebean.app.db.settings.settings.replace_if_newer_settings.Wifi;
import de.bikebean.app.db.sms.Sms;
import de.bikebean.app.db.type.types.SmsParserType;
import de.bikebean.app.ui.drawer.log.LogViewModel;

public class WifiOff extends SmsParserType {

    private static boolean matches() {
        return wifiStatusOffMatcher.find(0) && statusBatteryStatusMatcher.find(0);
    }

    public static @Nullable WifiOff createIfMatches(
            final @NonNull Sms sms,
            final @NonNull WeakReference<LogViewModel> lv) {
        wifiStatusOffMatcher = wifiStatusOffPattern.matcher(sms.getBody());
        statusBatteryStatusMatcher = statusBatteryStatusPattern.matcher(sms.getBody());

        if (matches())
            return new WifiOff(sms, lv);

        return null;
    }

    private WifiOff(final @NonNull Sms sms, final @NonNull WeakReference<LogViewModel> lv) {
        super(TYPE.WIFI_OFF, sms, lv);

        getSettings().add(new Battery(sms, super::getStatusBattery));
        getSettings().add(new Wifi(false, sms));
        getSettings().add(new Status(sms));
    }
}

File: app/src/main/java/de/bikebean/app/db/type/types/sms_parser_types/WifiOff.kt
package de.bikebean.app.db.type.types.sms_parser_types

import de.bikebean.app.db.settings.settings.add_to_list_settings.Battery
import de.bikebean.app.db.settings.settings.replace_if_newer_settings.Status
import de.bikebean.app.db.settings.settings.replace_if_newer_settings.Wifi
import de.bikebean.app.db.sms.Sms
import de.bikebean.app.db.type.types.ParserPatterns.statusBatteryStatusPattern
import de.bikebean.app.db.type.types.ParserPatterns.wifiStatusOffPattern
import de.bikebean.app.db.type.types.ParserType
import de.bikebean.app.ui.drawer.log.LogViewModel
import java.lang.ref.WeakReference

class WifiOff(sms: Sms, lv: WeakReference<LogViewModel>) : ParserType(TYPE.WIFI_OFF, sms, lv) {

    init {
        wifiStatusOffMatcher = wifiStatusOffPattern.matcher(sms.body)
        statusBatteryStatusMatcher = statusBatteryStatusPattern.matcher(sms.body)
    }

    override val matchers = listOf(
            wifiStatusOffMatcher,
            statusBatteryStatusMatcher
    )

    override val settings get() = listOf(
            Battery(sms) { statusBattery },
            Wifi(false, sms),
            Status(sms)
    )

}